internal/cmd/pagination.go

```diff
@@ -0,0 +1,50 @@
+package cmd
+
+import (
+	"github.com/infrahq/infra/api"
+	"github.com/infrahq/infra/internal/logging"
+)
+
+// listAll is a helper function that handles pagination and calls the given list request function.
+// listItems is the corresponding function in the API client that handles the Request "req".
+// handleError is a function that handles the error returned by the API client.
+func listAll[Item any, Req api.Paginatable](client *api.Client, req Req, listItems func(api.Client, Req) (*api.ListResponse[Item], error)) ([]Item, error) {
+
+	logging.Debugf("call server: page 1")
+
+	req, ok := req.SetPage(1).(Req)
+	if !ok {
+		panic("SetPage returned a different request type than expected")
+	}
+
+	res, err := listItems(*client, req)
+	if err != nil {
+		if api.ErrorStatusCode(err) == 403 {
+			logging.Debugf("%s", err.Error())
+			return nil, ErrMissingPrivileges
+		}
+		return nil, err
+	}
```

Comment:
Do we need to handle the error here? It seems like we return `err` in the default case anyway.

So I think we could avoid the need for `ErrMissingPrivileges` by always just returning `err`.

Reply:
The ErrMissingPrivileges exists to replicate the more user-friendly error messages that were originally in place for the 403 codes.

Reply:
The old logic should still continue to work without the new custom error. Does it not?